kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/status/NMStatusServiceImpl.java

```diff
@@ -29,14 +31,31 @@ public class NMStatusServiceImpl implements NetworkStatusService {
     private static final Logger logger = LoggerFactory.getLogger(NMStatusServiceImpl.class);
 
     private NMDbusConnector nmDbusConnector;
+    private NetworkService networkService;
+
+    public void setNetworkService(NetworkService networkService) {
+        this.networkService = networkService;
+    }
+
+    public NMStatusServiceImpl() {
+        try {
+            this.nmDbusConnector = NMDbusConnector.getInstance();
+        } catch (DBusExecutionException | DBusException e) {
+            logger.error("Cannot initialize NMDbusConnector due to: ", e);
+        }
+    }
+
+    public NMStatusServiceImpl(NMDbusConnector nmDbusConnector) {
+        this.nmDbusConnector = Objects.requireNonNull(nmDbusConnector);
+    }
 
     public void activate() throws DBusException {
         logger.debug("Activate NMStatusService...");
         this.nmDbusConnector = NMDbusConnector.getInstance();
```

Comment:
Since you acquire the NMDBusConnector instance in the constructor you don't need this in the `activate`

```suggestion
```